src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/bucketpath/Path.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.domain.bucketpath;
+
+public interface Path {
+
+    String name();
+
+    String separator();
```

Comment:
I feel methods in this abstraction doesn't take advantage of polymorphism. Because currently each subclass is more like data class without much different behavior.

I suggest we either move logic depending on it to each subclass or just use a data class for now:

```
class Path {
    name, separator, type ...
    getters ...
}
```

Reply:
Done.